src/world/Action/EffectBuilder.cpp

```diff
@@ -49,65 +42,79 @@ void EffectBuilder::moveToResultList( Entity::CharaPtr& chara, EffectResultPtr r
 {
-  EffectResultPtr nextResult = make_EffectResult( restoringTarget, getResultDelayMs() ); // restore mp source actor
+  EffectResultPtr nextResult = make_EffectResult( restoringTarget, nullptr, Common::Util::getTimeMs() + resultDelayMs );
   nextResult->restoreMP( amount, flag );
   moveToResultList( target, nextResult );
 }
 
-void EffectBuilder::damage( Entity::CharaPtr& effectTarget, Entity::CharaPtr& damagingTarget, uint32_t amount, Common::ActionHitSeverityType severity, Common::ActionEffectResultFlag flag )
+void EffectBuilder::damage( Entity::CharaPtr& effectTarget, Entity::CharaPtr& damagingTarget, uint32_t amount, Common::ActionHitSeverityType severity, Common::ActionEffectResultFlag flag, uint64_t resultDelayMs )
 {
-  EffectResultPtr nextResult = make_EffectResult( damagingTarget, getResultDelayMs() );
+  EffectResultPtr nextResult = make_EffectResult( damagingTarget, nullptr, Common::Util::getTimeMs() + resultDelayMs );
   nextResult->damage( amount, severity, flag );
   moveToResultList( effectTarget, nextResult );
 }
 
 void EffectBuilder::startCombo( Entity::CharaPtr& target, uint16_t actionId )
 {
-  EffectResultPtr nextResult = make_EffectResult( target, 0 );
+  EffectResultPtr nextResult = make_EffectResult( target, nullptr, 0 );
   nextResult->startCombo( actionId );
   moveToResultList( target, nextResult );
 }
 
 void EffectBuilder::comboSucceed( Entity::CharaPtr& target )
 {
-  EffectResultPtr nextResult = make_EffectResult( target, 0 );
+  EffectResultPtr nextResult = make_EffectResult( target, nullptr, 0 );
   nextResult->comboSucceed();
   moveToResultList( target, nextResult );
 }
 
-void EffectBuilder::applyStatusEffect( Entity::CharaPtr& target, uint16_t statusId, uint8_t param )
+void EffectBuilder::applyStatusEffect( Entity::CharaPtr& target, Entity::CharaPtr& source, uint16_t statusId, uint32_t duration, uint16_t param, uint64_t resultDelayMs )
+{
+  EffectResultPtr nextResult = make_EffectResult( target, source, Common::Util::getTimeMs() + resultDelayMs );
+  nextResult->applyStatusEffect( statusId, duration, param );
+  moveToResultList( target, nextResult );
+}
+
+void EffectBuilder::applyStatusEffect( Entity::CharaPtr& target, Entity::CharaPtr& source, StatusEffect::StatusEffectPtr pStatusEffect, uint64_t resultDelayMs )
+{
+  EffectResultPtr nextResult = make_EffectResult( target, source, Common::Util::getTimeMs() + resultDelayMs );
+  nextResult->applyStatusEffect( pStatusEffect );
+  moveToResultList( target, nextResult );
+}
+
+void EffectBuilder::statusNoEffect( Entity::CharaPtr& target, uint16_t statusId )
 {
-  EffectResultPtr nextResult = make_EffectResult( target, 0 );
-  nextResult->applyStatusEffect( statusId, param );
+  EffectResultPtr nextResult = make_EffectResult( target, nullptr, 0 );
+  nextResult->statusNoEffect( statusId );
   moveToResultList( target, nextResult );
 }
 
-void EffectBuilder::buildAndSendPackets()
+void EffectBuilder::buildAndSendPackets( float animationLock )
```

Comment:
again, why is this a parameter when it should just be a member var with a getter and setter?